### ISSUE TITLE
error handling

### ISSUE THREAD

**Issue by ctaggart** [ctaggart/autorust#26]
Do we wish to use a structured Error type like surf 2 moved to?
https://github.com/http-rs/surf/releases/tag/v2.0.0
https://blog.yoshuawuyts.com/async-http/#error-handling

**Comment by ctaggart** [comment]
Currently just using:
``` rust
pub type Error = Box<dyn std::error::Error + Send + Sync>;
pub type Result<T> = std::result::Result<T, Error>;
 
   Ok(res.json().await?)
```

In some use cases, it would be good to add more context, like the full response. Here I was using `anyhow` crate to achieve that.
``` rust
    let res = client.execute(req).await?;
    match res.error_for_status_ref() {
        Ok(_) => Ok(res.json().await?),
        Err(err) => {
            let e = Error::new(err);
            let e = e.context(res.text().await?);
            Err(e)
        }
    }
```

**Comment by ctaggart** [comment]
Planned implementation is listed in https://github.com/Azure/azure-sdk-for-rust/issues/47

**Comment by ctaggart** [comment]
fixed by #63
